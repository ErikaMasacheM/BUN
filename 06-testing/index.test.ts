import { describe, expect, test, beforeAll } from "bun:test";

beforeAll(() => {
    // setup tests
});

describe("math", () => {
    test("addition", () => {
        expect(2 + 2).toBe(4);
        expect(7 + 3).toMatchSnapshot();
    });

    test('test object snapshot', () => {
        const person = {
            name: 'Erika',
            age: 32
        }

        expect(person).toMatchSnapshot()
    });

    test('test object, expect data', () => {
        const person = {
            name: 'Erika',
            age: 32,
            id: 'uygtr5467gh'
        }

        expect(person).toEqual({
            name: 'Erika',
            age: 32,
            id: expect.any(String)
        });
    });
});
