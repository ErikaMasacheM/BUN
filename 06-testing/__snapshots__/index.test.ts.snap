// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`math addition 1`] = `10`;

exports[`math test object snapshot 1`] = `
{
  "age": 32,
  "name": "Erika",
}
`;
